tools/flaky_tests/check_branch.py

```diff
@@ -0,0 +1,142 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+""" This script is used to automate flaky test detection on CI
+
+This module calls each of the components to automate the detection of
+flaky tests. Currently, each test is given an equal number of runs
+such that all tests being checked can be run within the time budget.
+"""
+import logging
+import time
+import os
+import subprocess
+import json
+
+import flakiness_checker
+import diff_collator
+import dependency_analyzer
+
+LOGGING_FILE = os.path.join(os.path.dirname(__file__), "results.log")
+TESTS_DIRECTORY = "tests/python"
+PROFILING_TRIALS = 10
+TIME_BUDGET = 5400 
+
+logger = logging.getLogger(__name__)
+
+fh = logging.FileHandler(LOGGING_FILE)
+fh.setLevel(logging.INFO)
+fh.setFormatter(logging.Formatter("%(asctime)s - %(message)s"))
+logger.addHandler(fh)
+ 
+def select_tests(changes):
+    """returns tests that are dependent on given changes
+    """
+    top = subprocess.check_output(["git", "rev-parse", "--show-toplevel"])
+    top = top.decode("utf-8").splitlines()[0]
+    deps = dependency_analyzer.find_dependents(changes, top)
+
+    return [(filename, test) 
+            for filename in deps.keys() 
+            for test in deps[filename] 
+            if test.startswith("test_")]
```

Comment:
Make this "test_" a constant header and document this clearly.

Reply:
Done